src/main/scala/com.snowplowanalytics.snowplow.micro/ConfigHelper.scala

```diff
@@ -41,51 +45,69 @@ private[micro] object ConfigHelper {
+      igluConfigFile: Option[File] = None
     )
 
     val parser = new scopt.OptionParser[MicroConfig](buildinfo.BuildInfo.name) {
       head(buildinfo.BuildInfo.name, buildinfo.BuildInfo.version)
       help("help")
       version("version")
-      opt[File]("collector-config")
-        .required()
+      opt[Option[File]]("collector-config")
+        .optional()
         .valueName("<filename>")
         .text("Configuration file for collector")
-        .action((f: File, c: MicroConfig) => c.copy(collectorConfigFile = f))
+        .action((f: Option[File], c: MicroConfig) => c.copy(collectorConfigFile = f))
         .validate(f =>
-          if (f.exists) success
-          else failure(s"Configuration file $f does not exist"))
-      opt[File]("iglu")
-        .required()
+          f match {
+            case Some(file) =>
+              if (file.exists) success
+              else failure(s"Configuration file $f does not exist")
+            case None => success
+          }
+        )
+      opt[Option[File]]("iglu")
+        .optional()
         .valueName("<filename>")
         .text("Configuration file for Iglu igluClient")
-        .action((f: File, c: MicroConfig) => c.copy(igluConfigFile = f))
+        .action((f: Option[File], c: MicroConfig) => c.copy(igluConfigFile = f))
         .validate(f =>
-          if (f.exists) success
-          else failure(s"Configuration file $f does not exist"))
+          f match {
+            case Some(file) =>
+              if (file.exists) success
+              else failure(s"Configuration file $f does not exist")
+            case None => success
+          }
+        )
     }
-
+    
     val (collectorFile, igluFile) = parser.parse(args, MicroConfig()) match {
-      case Some(microConfig) =>
+      case Some(microConfig: MicroConfig) =>
         (microConfig.collectorConfigFile, microConfig.igluConfigFile)
       case None =>
         throw new RuntimeException("Problem while parsing arguments") // should never be called
     }
 
-    val collectorConfig = ConfigFactory.parseFile(collectorFile).resolve()
-    if (!collectorConfig.hasPath("collector"))
-      throw new IllegalArgumentException("Config file for collector doesn't contain \"collector\" path")
+    val resolved = collectorFile match {
+      case Some(f) => ConfigFactory.parseFile(f).resolve()
+      case None    => ConfigFactory.empty()
+    }
+
+    val collectorConfig = ConfigFactory.load(resolved.withFallback(ConfigFactory.load()))
 
-    val igluClient = getIgluClientFromFile(igluFile) match {
-      case Right(igluClient) => igluClient
-      case Left(e) =>
-        throw new IllegalArgumentException(s"Error while reading Iglu config file: $e.")
+    val igluClient = igluFile match {
+      case Some(f) =>
+        getIgluClientFromFile(f) match {
+          case Right(igluClient) => igluClient
+          case Left(e) =>
+            throw new IllegalArgumentException(s"Error while reading Iglu config file: $e.")
```

Comment:
I know this has nothing to do with any change you made, but this would be better as

```
throw new IllegalArgumentException("Error while reading Iglu config file", e)
```

where `e` is passed to the constructor [as the cause of the exception](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/IllegalArgumentException.html#%3Cinit%3E(java.lang.String,java.lang.Throwable)).

It might mean the user gets a slightly neater error message (although I haven't tested that).

Anyway, it's fine if you don't make this change for this release.

Reply:
I had a quick look at this, it seems like `e` is a `String` rather than `Throwable`. There's probably a way to get the throwable out of `getIgluClientFromFile`, but I'll leave it for now.

Reply:
Oh, my mistake, I misread it!  In that case, there's nothing wrong.  Sorry for false alarm.